docs/security_self_assessment.md

```diff
@@ -0,0 +1,1588 @@
+# CIS Benchmark Self Assessment Guide
+
+## Overview
+
+This document is a companion to the security [hardening guide](hardening_guide.md). The hardening guide provides prescriptive guidance for hardening a production installation of RKE2, and this benchmark guide is meant to help you evaluate the level of security of the hardened cluster against each control in the benchmark.
+
+This document is to be used by Rancher operators, security teams, auditors and decision makers.
+
+For more detail about each audit, including rationales and remediations for failing tests, you can refer to the corresponding section of the CIS Kubernetes Benchmark v1.5. You can download the benchmark after logging in to [CISecurity.org](https://www.cisecurity.org/benchmark/kubernetes/).
+
+## Testing Controls Methodology
+
+RKE2 installs Kubernetes services via static pods. Configuration is defined by arguments passed to the pod at the time of initialization.
+
+Where control audits differ from the original CIS benchmark, the audit commands specific to RKE2 are provided for testing. When performing the tests, you will need access to the command line on the hosts of the RKE2 nodes. The commands also make use of the the `jq` and `kubectl` (with valid config) tools to and are required in the testing and evaluation of test results.
+
+## Controls
+
+### 1 Master Node Security Configuration
+
+#### 1.1 Master Node Configuration Files
+
+*1.1.1* Ensure that the API server pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
```

Comment:
I think all these config file controls _are_ applicable. We have them in rke2.

For example:
```
ls -al  /var/lib/rancher/rke2/agent/pod-manifests/kube-apiserver.yaml
-rw-r--r-- 1 root root 6692 Sep 23 03:15 /var/lib/rancher/rke2/agent/pod-manifests/kube-apiserver.yaml
```

Reply:
As I understand it, the pod manifest contains the cli args to pass and doesn't use a config file. The process itself is then looked at for the values further down.